frontend/src/workbench/WorkbenchMachine.ts

```diff
@@ -20,21 +20,25 @@ export interface WorkbenchStateSchema {
 export type SchemaValue = 'initial';
 
 export interface WorkbenchContext {
-  selection: Selection | null;
+  selection: Selection;
   representations: Representation[];
   displayedRepresentation: Representation | null;
 }
 
 export type ShowRepresentationEvent = { type: 'SHOW_REPRESENTATION'; representation: Representation };
 export type HideRepresentationEvent = { type: 'HIDE_REPRESENTATION'; representation: Representation };
-export type UpdateSelectionEvent = { type: 'UPDATE_SELECTION'; selection: Selection; isRepresentation: boolean };
+export type UpdateSelectionEvent = {
+  type: 'UPDATE_SELECTION';
+  selection: Selection;
+  representations: Representation[];
+};
 export type WorkbenchEvent = UpdateSelectionEvent | ShowRepresentationEvent | HideRepresentationEvent;
 
 export const workbenchMachine = Machine<WorkbenchContext, WorkbenchStateSchema, WorkbenchEvent>(
   {
     initial: 'initial',
     context: {
-      selection: null,
+      selection: { entries: [] },
```

Comment:
Now that the object `selection` on which everything is connected cannot be `null` anymore, there should not be any occurrences of `selection?` in the code again. There are still some in your PR.

Reply:
Done